prerequisites/build-functions/patches/gcc/8.1.0/partially-revert-rev259385.diff

```diff
@@ -0,0 +1,61 @@
+diff --git a/gcc/fortran/dependency.c b/gcc/fortran/dependency.c
+index 3e14ddc25d8..a0bbd584947 100644
+--- a/gcc/fortran/dependency.c
++++ b/gcc/fortran/dependency.c
+@@ -2238,9 +2238,8 @@ gfc_dep_resolver (gfc_ref *lref, gfc_ref *rref, gfc_reverse *reverse)
+ 	    break;
+ 
+ 	  /* Exactly matching and forward overlapping ranges don't cause a
+-	     dependency, when they are not part of a coarray ref.  */
+-	  if (fin_dep < GFC_DEP_BACKWARD
+-	      && lref->u.ar.codimen == 0 && rref->u.ar.codimen == 0)
++	     dependency.  */
++	  if (fin_dep < GFC_DEP_BACKWARD)
+ 	    return 0;
+ 
+ 	  /* Keep checking.  We only have a dependency if
+diff --git a/gcc/fortran/trans-intrinsic.c b/gcc/fortran/trans-intrinsic.c
+index 00edd447bb2..87b3ca72c05 100644
+--- a/gcc/fortran/trans-intrinsic.c
++++ b/gcc/fortran/trans-intrinsic.c
+@@ -1860,7 +1860,7 @@ conv_caf_send (gfc_code *code) {
+ 
+   lhs_expr = code->ext.actual->expr;
+   rhs_expr = code->ext.actual->next->expr;
+-  may_require_tmp = gfc_check_dependency (lhs_expr, rhs_expr, false) == 0
++  may_require_tmp = gfc_check_dependency (lhs_expr, rhs_expr, true) == 0
+ 		    ? boolean_false_node : boolean_true_node;
+   gfc_init_block (&block);
+ 
+diff --git a/gcc/testsuite/gfortran.dg/coarray_dependency_1.f90 b/gcc/testsuite/gfortran.dg/coarray_dependency_1.f90
```

Comment:
Unless we anticipate running the GCC test-suite, we may want to remove this line (L30) to the end off the patch. The less hunks we are patching the less likely the application will fail. Just an idea, take it or leave it.

Reply:
@zbeekman Great idea. Feel free to push a commit that removes that part of the patch or let me know if I should do so.